packages/module/src/FieldBuilder/FieldBuilder.tsx

```diff
@@ -0,0 +1,330 @@
+   */
+  rowGroupLabelPrefix?: string;
+  /**
+   * Optional unique ID prefix for this FieldBuilder instance.
+   * This ensures unique IDs when multiple FieldBuilders exist on the same page.
+   */
+  fieldBuilderIdPrefix?: string;
+}
+
+/**
+ * FieldBuilder is a component group that simplifies the creation of dynamic,
+ * multi-row forms with a consistent layout. It manages the layout and actions
+ * for adding and removing rows, while giving the consumer full control over the fields themselves.
+ */
+export const FieldBuilder: FunctionComponent<FieldBuilderProps> = ({
+  firstColumnLabel,
+  secondColumnLabel,
+  rowCount,
+  children,
+  onAddRow,
+  onRemoveRow,
+  addButtonProps = {},
+  rowGroupLabelPrefix = "Row",
+  fieldBuilderIdPrefix = "field-builder",
+  ...formGroupProps
+}: FieldBuilderProps) => {
+  // Track the previous row count to detect when rows are added
+  const prevRowCountRef = useRef(rowCount);
+  // Track focusable elements for each row
+  const focusableElementsRef = useRef<Map<number, HTMLElement>>(new Map());
+  // Track remove button refs for focus management
+  const removeButtonRefs = useRef<Map<number, HTMLButtonElement>>(new Map());
+  // Track the add button ref
+  const addButtonRef = useRef<HTMLButtonElement>(null);
+  // ARIA live region for announcing dynamic changes
+  const liveRegionRef = useRef<HTMLDivElement>(null);
+
+  // Generate unique IDs for this instance
+  const instanceId = useRef(`${fieldBuilderIdPrefix}-${Math.random().toString(36).substr(2, 9)}`);
+  const firstColumnLabelId = `${instanceId.current}-first-column-label`;
+  const secondColumnLabelId = secondColumnLabel ? `${instanceId.current}-second-column-label` : undefined;
+
+  // Function to announce changes to screen readers
+  const announceChange = useCallback((message: string) => {
+    if (liveRegionRef.current) {
+      liveRegionRef.current.textContent = message;
+      // Clear the message after a delay to prepare for next announcement
+      setTimeout(() => {
+        if (liveRegionRef.current) {
+          liveRegionRef.current.textContent = '';
+        }
+      }, 1000);
+    }
+  }, []);
+
+  // Create ref callback for focusable elements
+  const createFocusRef = useCallback((rowIndex: number) => 
+    (element: HTMLElement | null) => {
+      if (element) {
+        focusableElementsRef.current.set(rowIndex, element);
+      } else {
+        focusableElementsRef.current.delete(rowIndex);
+      }
+    }, []);
+
+  // Create ref callback for remove buttons
+  const createRemoveButtonRef = useCallback((rowIndex: number) => 
+    (element: HTMLButtonElement | null) => {
+      if (element) {
+        removeButtonRefs.current.set(rowIndex, element);
+      } else {
+        removeButtonRefs.current.delete(rowIndex);
+      }
+    }, []);
+
+  // Enhanced onAddRow with focus management and announcements
+  const handleAddRow = useCallback(() => {
+    onAddRow();
```

Comment:
Depending on the resolution of the above comment, either in the component code or example code, when a new row is added we should shift focus to that new rows first input.